src/java.desktop/share/classes/javax/swing/JScrollBar.java

```diff
@@ -753,11 +753,41 @@ public void stateChanged(ChangeEvent e)   {
         }
     }
 
-    // PENDING(hmuller) - the next three methods should be removed
+    /**
+     * Unlike most components, {@code JScrollBar} derives the minimum size from
+     * the preferred size in one axis and a fixed minimum size in the other.
+     * Thus, it overrides {@code JComponent.setMinimumSize} contract
+     * that subsequent calls to {@code getMinimumSize} will return the
+     * same value as set in {@code JComponent.setMinimumSize}.
+     *
+     * @param minimumSize the new minimum size of this component
+     */
+    public void setMinimumSize(Dimension minimumSize) {
+        super.setMinimumSize(minimumSize);
+    }
 
     /**
-     * The scrollbar is flexible along it's scrolling axis and
+     * Unlike most components, {@code JScrollBar} derives the maximum size from
+     * the preferred size in one axis and a fixed maximum size in the other.
+     * Thus, it overrides {@code JComponent.setMaximumSize} contract
+     * that subsequent calls to {@code getMaximumSize} will return the
+     * same value as set in {@code JComponent.setMaximumSize}.
+     *
+     * @param maximumSize the desired maximum allowable size
+     */
+    public void setMaximumSize(Dimension maximumSize) {
+        super.setMaximumSize(maximumSize);
+    }
+
+    /**
+     * Returns the minimum size for the {@code JScrollBar}.
+     * The scrollbar is flexible along its scrolling axis and
      * rigid along the other axis.
```

Comment:
I would rather clarify this property by describing how the returned value is derived from the preferred size. Does it make sense? Should it be an `@implNote` rather than a spec? Or is it completely unnecessary?

Reply:
Not sure..Since it is already reviewed and approved by Phil, I chose to keep it same..

Reply:
This change requires a CSR which hasn't been approved yet. It's an opportunity to improve the specification for `getMinimumSize` and `getMaximumSize`.

After the CSR is approved, further updates will be more of a hassle.

This is why I'm asking these questions here. “No, we don't want to update,” is also a valid answer. Let's see what @prrace has to say.